{
   "active_lock_reason" : null,
   "assignee" : null,
   "assignees" : [],
   "author_association" : "NONE",
   "body" : "<!-- Describe the issue -->\r\nIssue:\r\n\r\nThe Checksum generated by `getdescriptorinfo` is not accepted when I use it in `importdescriptors`. The two RPC calls calculate a different checksum for the same descriptor. \r\n\r\n<!--- What behavior did you expect? -->\r\n\r\nI ran the following commands on MacOS with bitcoind v0.21.1 installed. Since my descriptor lacked a checksum, I used `getdescriptorinfo` to generate one, as I read here: https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md#checksums\r\n```\r\nThe checksum for a descriptor without one can be computed using the getdescriptorinfo RPC.\r\n```\r\n\r\nI expected the checksum generated by `getdescriptorinfo` to be accepted by `importdescriptors`. It was not. \r\n\r\n<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->\r\n\r\nThe actual behavior is below. I am redacting the actual descriptors/hashes for personal privacy.\r\n\r\n```bash\r\n$ bitcoin-cli --version\r\nBitcoin Core RPC client version v0.21.1\r\n# getdescriptorinfo generates a checksum beginning with 3a.\r\n$ bitcoin-cli getdescriptorinfo \"<DESCRIPTOR>\"\r\n{\r\n  \"descriptor\": \"<DESCRIPTOR>#3a...\",\r\n  \"checksum\": \"3a...\",\r\n  \"isrange\": false,\r\n  \"issolvable\": true,\r\n  \"hasprivatekeys\": false\r\n}\r\n# importdescriptors rejects the checksum generated by getdescriptorinfo and calculates its own different one\r\n$ bitcoin-cli -rpcwallet=my_wallet importdescriptors '[{ \"desc\": \"<DESCRIPTOR>#3a...\", \"timestamp\": \"now\" }]'\r\n[\r\n  {\r\n    \"success\": false,\r\n    \"error\": {\r\n      \"code\": -5,\r\n      \"message\": \"Provided checksum '3a...' does not match computed checksum '6z...'\"\r\n    }\r\n  }\r\n]\r\n# importdescriptors properly accepts the checksum provided by itself.\r\n$ bitcoin-cli -rpcwallet=my_wallet importdescriptors '[{ \"desc\": \"<DESCRIPTOR>#6z...\", \"timestamp\": \"now\" }]' \r\n[\r\n  {\r\n    \"success\": true\r\n  }\r\n]\r\n# Using checksum expected by importdescriptors throws an error when used in getdescriptorinfo cmd\r\n$ bitcoin-cli getdescriptorinfo \"<DESCRIPTOR>#6z...\"\r\nerror code: -5\r\nerror message:\r\nProvided checksum '6z40f68x' does not match computed checksum '3av8evsc'\r\n```\r\n\r\n**System information**\r\n\r\n<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->\r\nBitcoin Core info:\r\nBitcoin Core RPC client version v0.21.1 Downloaded from Bitcoin.org binary. Not compiled from source.\r\n\r\n<!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->\r\nMachine/OS: \r\nMacOS Catalina 10.15.6 \r\nMemory 16GB\r\n",
   "closed_at" : "2021-11-17T20:34:44Z",
   "closed_by" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/46731461?v=4",
      "events_url" : "https://api.github.com/users/SachinMeier/events{/privacy}",
      "followers_url" : "https://api.github.com/users/SachinMeier/followers",
      "following_url" : "https://api.github.com/users/SachinMeier/following{/other_user}",
      "gists_url" : "https://api.github.com/users/SachinMeier/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/SachinMeier",
      "id" : 46731461,
      "login" : "SachinMeier",
      "node_id" : "MDQ6VXNlcjQ2NzMxNDYx",
      "organizations_url" : "https://api.github.com/users/SachinMeier/orgs",
      "received_events_url" : "https://api.github.com/users/SachinMeier/received_events",
      "repos_url" : "https://api.github.com/users/SachinMeier/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/SachinMeier/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/SachinMeier/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/SachinMeier"
   },
   "comments" : 7,
   "comments_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539/comments",
   "created_at" : "2021-11-17T20:06:43Z",
   "events_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539/events",
   "html_url" : "https://github.com/bitcoin/bitcoin/issues/23539",
   "id" : 1056584783,
   "labels" : [
      {
         "color" : "FBBAAB",
         "default" : false,
         "description" : null,
         "id" : 64585,
         "name" : "Bug",
         "node_id" : "MDU6TGFiZWw2NDU4NQ==",
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/labels/Bug"
      }
   ],
   "labels_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539/labels{/name}",
   "locked" : false,
   "milestone" : null,
   "node_id" : "I_kwDOABII584--jRP",
   "number" : 23539,
   "performed_via_github_app" : null,
   "reactions" : {
      "+1" : 0,
      "-1" : 0,
      "confused" : 0,
      "eyes" : 0,
      "heart" : 0,
      "hooray" : 0,
      "laugh" : 0,
      "rocket" : 0,
      "total_count" : 0,
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539/reactions"
   },
   "repository_url" : "https://api.github.com/repos/bitcoin/bitcoin",
   "state" : "closed",
   "timeline_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539/timeline",
   "title" : "Descriptor Checksum Inconsistency",
   "updated_at" : "2021-11-17T20:35:59Z",
   "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/23539",
   "user" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/46731461?v=4",
      "events_url" : "https://api.github.com/users/SachinMeier/events{/privacy}",
      "followers_url" : "https://api.github.com/users/SachinMeier/followers",
      "following_url" : "https://api.github.com/users/SachinMeier/following{/other_user}",
      "gists_url" : "https://api.github.com/users/SachinMeier/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/SachinMeier",
      "id" : 46731461,
      "login" : "SachinMeier",
      "node_id" : "MDQ6VXNlcjQ2NzMxNDYx",
      "organizations_url" : "https://api.github.com/users/SachinMeier/orgs",
      "received_events_url" : "https://api.github.com/users/SachinMeier/received_events",
      "repos_url" : "https://api.github.com/users/SachinMeier/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/SachinMeier/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/SachinMeier/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/SachinMeier"
   }
}
